OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -251,26 +251,29 @@ protected virtual bool CanFire(Actor self, Target target)
 
 		// Note: facing is only used by the legacy positioning code
 		// The world coordinate model uses Actor.Orientation
-		public virtual Barrel CheckFire(Actor self, IFacing facing, Target target)
+		public virtual Barrel[] CheckFire(Actor self, IFacing facing, Target target)
 		{
-			if (!CanFire(self, target))
-				return null;
+			var barrels = new List<Barrel>();
 
-			if (ticksSinceLastShot >= Weapon.ReloadDelay)
-				Burst = Weapon.Burst;
+			while (CanFire(self, target))
```

Comment:
This will loop infinitely if the weapon has `BurstDelays: 0`, doesn't use an ammopool to pause the trait, and is in range and valid against the target.